core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -145,36 +145,29 @@ class KafkaZkClient private[zk] (zooKeeperClient: ZooKeeperClient,
     }.toMap
   }
 
-  def getFederatedTopic(topic: String): String = {
-    val getDataRequest = GetDataRequest(FederatedTopicZnode.path(topic))
-    val getDataResponse = retryRequestUntilConnected(getDataRequest)
-    getDataResponse.resultCode match {
-      case Code.OK => "/" + FederatedTopicZnode.decode(getDataResponse.data) + "/" + topic
-      case Code.NONODE => null
-      case _ => throw getDataResponse.resultException.get
+  def getFederatedTopic(topic: String, namespace: String): String = {
+    if (pathExists(FederatedTopicZnode.path(topic, namespace))) {
+      "/" + namespace + "/" + topic
+    } else {
+      null
```

Comment:
Idiomatic way in scala (assuming this `KafkaZkClient` class is not used by anything other than kafka) is to use `Option` instead `of null`.

so make it return Option[String] via
```scala
if (...) {
  Some(s"/.....")
} else {
  None
}
```

Reply:
Also to ensure that it's not used elsewhere, this method can become 
```scala
// scoped on the KafkaApi Class
private[KafkaApi] def getFederatedTopic(topic: String, namespace: String): String = {
```
or 
```scala
// scoped the server package
private[server] def getFederatedTopic(topic: String, namespace: String): String = {
```